source/blog/2022-11-17-announcing-hanami-200rc.html.markdown

```diff
@@ -0,0 +1,357 @@
+
+something something also helps you separate deps from other arguments?
+
+The Deps mixin makes object composition natural and low-friction, making it easy for you to break down unwieldy classes. Before long you’ll be creating better single-responsibility and reusable components, easier to test and easier to understand. Once you get started, you’ll never want to go back!
+
+### Blazing fast new router
+
+Any web app needs a way to let users in, and Hanami 2.0 offers a friendly, intuitive routing DSL. You can add your routes to `config/routes.rb`:
+
+```ruby
+module Bookshelf
+  class Routes < Hanami::Routes
+    root to: "home.show"
+
+    get "/books", to: "books.index"
+    get "/books/:slug", to: "books.show"
+    post "/books/:slug/reviews", to: "books.comments.create"
+  end
+end
+```
+
+We’ve completely rewritten the router’s engine, with benchmarks showing it [outperforms nearly all others](https://hanamirb.org/blog/2020/02/26/introducing-hanami-api/).
+
+### Redesigned action classes
+
+From routes we move to actions, the classes for handling individual HTTP endpoints. In Hanami 2.0 we’ve redesigned actions to fit seamlessly with the rest of your app.
+
+In your actions you can now `include Deps` like any other class, which makes it to keep your business logic separate and your actions focused on HTTP interactions only:
+
+```ruby
+module Bookshelf
+  module Actions
+    module Books
+      class Show < Bookshelf::Action
+        include Deps["book_repo"]
+
+        params do
+          required(:slug).value(:string)
+        end
+
+        def handle(request, response)
+          book = book_repo.find_by_slug(request.params[:slug])
+
+          response.body = book.to_json
+        end
+      end
+    end
+  end
+end
+```
+
+Actions in Hanami still provide everything you need for a full-featured HTTP layer, including built-in parameter validation, Rack integration, session and cookie handling, flash messages, before/after callbacks and more.
+
+### Type-safe app settings
+
+For your app to do its thing, you'll want to give it various settings: behavioral toggles, API keys, external system URLs and the like. Hanami 2.0 provides a built-in settings class along with [dotenv][dotenv] integration for loading these settings from `.env*` files in local development.
+
+You can define your settings along with type constructors in `config/settings.rb`:
+
+```ruby
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :email_client_api_key, constructor: Types::String
+    setting :emails_enabled, constructor: Types::Params::Bool
+  end
+end
+```
+
+Your settings are then available as a `"settings"` component for you to use from anywhere in your app:
+
+```ruby
+# Given ENV["EMAILS_ENABLED"] = "true"
+Hanami.app["settings"].emails_enabled # => true
```

Comment:
Could we use a convention that we use symbols for built-in system-level components? so `Hanami.app[:logger]` `Hanami.app[:settings]` and `Hanami.app[:notifications]`?